examples/rvv_reduce.c

```diff
@@ -30,9 +30,9 @@ void reduce(double *a, double *b, double *result_sum, int *result_count,
     vfloat64m1_t vec_a = __riscv_vle64_v_f64m1(a, vl);
     vfloat64m1_t vec_b = __riscv_vle64_v_f64m1(b, vl);
 
-    vbool64_t mask = __riscv_vmfne_vv_f64m1_b64(vec_a, vec_zero, vl);
+    vbool64_t mask = __riscv_vmfne_vf_f64m1_b64(vec_a, 0, vl);
 
-    vec_s = __riscv_vfmacc_vv_f64m1_m(mask, vec_s, vec_a, vec_b, vl);
+    vec_s = __riscv_vfmacc_vv_f64m1_tu(vec_s, vec_a, vec_b, vl);
```

Comment:
Does this change the behavior?

Reply:
Yes, but in a good way:

If we want to use a vector register to collect all sums and reduce afterward, we need to make sure the tail policy is set to undisturbed, otherwise the last iteration may overwrite the inactive elements.
The mask is redundant, because multiplying by zero results in zero.

Reply:
I see your point. My intent was to demonstrate the masked instructions though, lets change the if statement to checking something that is not zero?

Reply:
Let's change the name to `reduce_vec` too?